src/test/kotlin/com/bardsoftware/backend/fleet/rmarkdown/PandocTest.kt

```diff
@@ -40,10 +39,12 @@ class PandocTest {
         val markdown = Paths.get("src","test","resources",rootFileName).toString()
         val outputName = Files.getNameWithoutExtension(markdown) + ".tex"
 
-        val mockConfig = mock(Config::class.java)
-        `when`(mockConfig.getString("pandoc.compile.command")).thenReturn(CP_COMMAND)
+        val mockConfig = ConfigFactory
+                .empty()
+                .withValue("pandoc.compile.command", ConfigValueFactory.fromAnyRef(CP_COMMAND))
+        val cpArguments = listOf("source", "dest")
```

Comment:
If you want to be able to use different command line depending on the config with the same binary then you obviously need to have the same names of arguments. You can't replace command line 
```
launch-pandoc \\${projectRootAbsPath} \\${workingDirRelPath} \\${inputFileName} \\${outputFileName} \\${mainFont}" 
```
with `cp \${source} \${dest}` without recompiling the binary -- your arguments have different names.

Reply:
Not literally to config file, because that will make test code more complicated. You already  create Config with this cp command in the test, it is the same as if it was read from file. But the fact that you use different arg names makes your test testing itself instead of testing production. We want to suppy a command with placeholders to production, and be sure that production code will fill the placeholders properly, no matter if the command is `cp`, `pandoc`, `docker run` or whatever else.